src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,195 @@
+    private T item;
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    {
+        Node<T> head = null;
+        Node<T> tail = null;
+    }
+
+    public MyLinkedList(T item) {
+        this.item = item;
+    }
+
+    public MyLinkedList(){
+    }
+
+    public void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("exception");
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = node;
+            head.prev = null;
+            tail.next = null;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            tail = node;
+            tail.next = null;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("exception");
+        }
+        if (size > 0 && index < (size - 1) && index > 0) {
+            Node<T> node = new Node<>(head, value, tail);
+            for (int i = 1; i < index; i++) {
+                node.prev = node.prev.next;
+            }
+            node.prev.next = node;
+            for (int i = 1; i < size - index; i++) {
+                node.next = node.next.prev;
+            }
+            node.next.prev = node;
+            size++;
+        }
+        if ((size - 1) == index) {
+            Node<T> node = new Node<>(null, value, null);
+            node.prev = tail.prev;
+            tail.prev = node;
+            node.next = tail;
+            node.prev.next = node;
+            size++;
+        }
+        if (index == 0 && size > 0) {
+            Node<T> node = new Node<>(null, value, null);
+            head.prev = node;
+            node.next = head;
+            head = node;
+            head.prev = null;
+            size++;
+        }
+        if (index == size) {
+            add(value);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
+        checkIndex(index);
+        if (size == 0) {
+            return null;
+        }
+        if (index == 0) {
+            return head.item;
+        }
+        if (index > 0) {
+            Node<T> currentNode = head;
+            for (int i = 0; i < index; i++) {
+                currentNode = currentNode.next;
+            }
+            return currentNode.item;
+        }
         return null;
```

Comment:
Please make this method separate and reuse it in next methods.

"Don’t create repeating code.
If the logic of your code repeats - move it to the separate private method. Remember about DRY and KISS principles."